eZ/Publish/Core/Persistence/Cache/UserHandler.php

```diff
@@ -339,4 +349,21 @@ public function unAssignRole($contentId, $roleId)
 
         return $return;
     }
+
+    /**
+     * @see eZ\Publish\SPI\Persistence\User\Handler::removeRoleAssignment
+     */
+    public function removeRoleAssignment($roleAssignmentId)
+    {
+        $this->logger->logCall(__METHOD__, array('assignment' => $roleAssignmentId));
+        $return = $this->persistenceHandler->userHandler()->removeRoleAssignment($roleAssignmentId);
+
+        // We don't know the contentId, so clear all assignment cache.
+        // TODO: fetch contentId to avoid this?
+        $this->cache->clear('user', 'role', 'assignments');
+        // TODO: Clear this too/instead? Fetch role from assignment and clear by roleId?
```

Comment:
I think the above is sufficient, there is no need to clear everything for the user/role.